lua/luasnip/nodes/choiceNode.lua

```diff
@@ -45,7 +49,47 @@ function ChoiceNode:init_nodes()
 	self.active_choice = self.choices[1]
 end
 
-local function C(pos, choices, opts)
+---@class LuaSnip.Opts.ChoiceNode: LuaSnip.Opts.Node
+---@field restore_cursor boolean? If set, the currently active node is looked up
+---in the switched-to choice, and the cursor restored to  preserve the current
+---position relative to that node. The node may be found if a `restoreNode` is
+---present in both choice.
+---`false` by default, as enabling might lead to decreased performance.
+---
+---It's possible to override the default by wrapping the `choiceNode`
+---constructor in another function that sets `opts.restore_cursor` to `true` and
+---then using that to construct `choiceNode`s:
+---```lua
+---local function restore_cursor_choice(pos, choices, opts)
+---    if opts then
+---        opts.restore_cursor = true
+---    else
+---        opts = {restore_cursor = true}
+---    end
+---    return c(pos, choices, opts)
+---end
+---```
+---Consider passing this override into `snip_env`.
+---@field node_callbacks table<"change_choice"|"enter"|"leave", fun(node:LuaSnip.Node)>
+---Specify functions to call after changing the choice, or entering or leaving
+---the node. The callback receives the `node` the callback was called on.
+
+---Create a new choiceNode.
+---@param pos integer Jump-index of the node (See
+---[Basics-Jump-Index](../../../DOC.md#jump-index)).
```

Comment:
Could we avoid relative path like this and instead work from root of repo, thus links would be only `DOC.md#jump-index` ?
We could even simplify to `#jump-index` since everything will be in the same file 🤔

Reply:
> and one could actually jump to it, if the editor supports it

I didn't think of this, it's actually a good idea 👍